userd/autostart.go

```diff
@@ -35,10 +34,25 @@ import (
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
 	"github.com/snapcore/snapd/strutil/shlex"
 	"github.com/snapcore/snapd/systemd"
 )
 
+var (
+	currentDesktop = splitSkippingEmpty(os.Getenv("XDG_CURRENT_DESKTOP"), ":")
```

Comment:
this  function stuck in my craw, and I needed to come back here to say: dude, that's `strings.FieldsFunc(s, func(r rune) bool { r ==  substr })`, given everywhere you use it `substr` is just a byte :-)

Reply:
the _reason_ it stuck was because you're doing double the allocations that you need to do, and you don't control the thing you're splitting so it can be arbitrarily big (up to the limit of an environment variable, which is a lot bigger than you might expect)

Reply:
Thanks for the suggestions!

Reply:
> Do we need to be case-sensitive here btw?
I honestly have no idea. The spec https://standards.freedesktop.org/desktop-entry-spec/latest/ar01s06.html is not very helpful either. However, I"ve seen desktop files using `OnlyShowIn=GNOME;Unity;`, also AFAIK GNOME and set `XDG_CURRENT_DESKTOP` to `GNOME` and `Unity` respectively, I'd say this is case sensitive.